### PR TITLE
Generic Table

### LINKED ISSUE #19: Create generic table

### PR DESCRIPTION
**Pull request by ketphan02**
Closes #19 

Shadcn components (not modified):
- Input
- Select
- Table
- Dropdown Menu
- Data Table Pagination (https://ui.shadcn.com/docs/components/data-table#pagination-1)

Modified components (need review)
- Data Table Column Header (modified from https://ui.shadcn.com/docs/components/data-table#column-header)
- SearchBar
- **Data Table**:
  - This table can sort/filter/search
  - This table can also have customizable components on top right
  - Have pagination
  - Have row action
  - Have general action
  - Have bulk action

### CODE REVIEW

File: src/components/ui/data-table-column-header.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import React from "react"
+
+interface DataTableColumnHeaderProps<TData, TValue>
+    extends React.HTMLAttributes<HTMLDivElement> {
+    column: Column<TData, TValue>
+    title: string
+    hasDropDownMenu?: boolean
+}
+
+export function DataTableColumnHeader<TData, TValue>({column, title, className, hasDropDownMenu = true}: DataTableColumnHeaderProps<TData, TValue>) {
```
Comment (abeeto) [inline src/components/ui/data-table-column-header.tsx:30]: awesome, thanks for modifying component so that columns have option for no dropdown

**Comment by Opeyem1a** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #20** <a href="https://app.graphite.dev/github/pr/Teamable-Analytics/teamable-frontend/20?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/Teamable-Analytics/teamable-frontend/20?utm_source=stack-comment).

**Review by justino599 — APPROVED** [review]
wow he really cooked that hard

**Review by SanaBob — APPROVED** [review]
nice pr opey

**Comment by abeeto** [comment]
lgtm

**Pull request merged**